README.md

```diff
@@ -0,0 +1,197 @@
+
+2. Create a U16 FIFO for transmitted data.
+
+3. Create a custom type ModbusTxInstruction FIFO for internal usage while
+   transmitting data. Specify never arbitrate for read and write.
+
+4. Create two custom type HealthAndStatusUpdate FIFOs - one for receiving,
+   second for transmission.
+
+5. Create a boolean register for IRQ control and "WaitForRXFrame". Specify
+   never arbitrate for write.
+
+6. Generate a boolean register for the software controlled trigger. Specify
+   never arbitrate for write. You can use a single trigger for controlling
+   multiple ports so this is an optional step if a register has already been
+   generated.
+
+7. Create a FIFO for Tx timestamps. Specify a data type of U64. Specify never
+   arbitrate for write.
+
+8. Drop the FPGAModbus/ModbusPort.vi onto your main VI. Setup the port settings
+   using the resources defined above. Sets base address (shall be multiply of
+   6) and IRQ (ordinals).  Review the comments in ModbusPort.vi for more
+   information.
+
+### Optional
+
+To copy health and status data FIFOs into a single FIFO:
+
+9. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+   health and status FIFO. Set the source as the FIFO generated in step 2.
+
+10. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+    health and status FIFO. Set the source as the FIFO generated in step 5.
+
+# Theory of operation
+
+FIFOs are used for what they are best suited - to provide a channel between
+variable speeds write and read ends. Usually one end of the FIFO is inside
+[SCTL](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000P8sWSAS&l=en-US).
+SCTL are used to grant execution of the commands inside FPGA in a single tick.
+
+Link speed is hard-coded to 44 40Mhz ticks per bit, which roughly equal to
+921600 bauds *(actual number is 40M / 44 = 909090.909; but as delays are
+inserted between frames, and frames aren't long, 44 works, sampling a slightly
+different part of receiving signal for each received bit)*.
+
+One start and one stop bits are written (must be included in data, see below).
+
+Transmitting FIFO is U16 typed. The most significant nibble is used for
+instruction. Nibbles 2-4 are used for data.
+
+Data from transmitting FIFO are parsed into instructions and data, and placed
+into internal FIFO. Instructions are then processed from the internal FIFO -
+see below for instructions types. Proper signal (frame start, start bit, data,
+stop bit, frame delay) are generated.
+
+Received line is checked for frame start. Bytes are then stored into receiving
+FIFO. Frame end is detected on line silence. IRQ is raised when frame is
+received. As the expected use is in a master-slave communication, IRQ doesn't
+have to be handled outside ModbusPort subVI. Data are writen into receiving
+FIFO.
+
+CRC16 isn't checked on received FIFO and the CRC16 (if part of the
+communication) is stored into receiving FIFO.
```

Comment:
Then why mention CR16 at all? It appears that the FIFO doesn't understand or make use of it.